### ISSUE TITLE
CI Integration

### ISSUE THREAD

**Issue by axegon** [axegon/spectrust#1]

**Comment by KarnikaSaxena011** [comment]
ImportError: cannot import name 'generate_spectrogram' from 'spectrust._spectro' (unknown location)
How to resolve this issue?

**Comment by axegon** [comment]
@KarnikaSaxena011 how did you install the package? Also what are you trying to import?

**Comment by KarnikaSaxena011** [comment]
> @KarnikaSaxena011 how did you install the package? Also what are you trying to import?

I am trying to import spectrust.
Import spectrust
But it is throwing error and the platform i am using is vs code.

**Comment by KarnikaSaxena011** [comment]
> @KarnikaSaxena011 how did you install the package? Also what are you trying to import?

Although there is no such issue if i use google colab.

**Comment by KarnikaSaxena011** [comment]
> @KarnikaSaxena011 how did you install the package? Also what are you trying to import?

I dowloaded your source code and put it in my project folder.
But i am geeting this error:- ImportError: cannot import name 'generate_spectrogram' from 'spectrust._spectro' (unknown location)

**Comment by axegon** [comment]
That's not going to work, the underlying implementation is in rust, therefore you'll need to compile it. Long story-short:

1. [Install rust](https://www.rust-lang.org/tools/install)(and git if you haven't).
2. `pip install git+https://github.com/axegon/spectrust.git`

**Comment by KarnikaSaxena011** [comment]
> That's not going to work, the underlying implementation is in rust, therefore you'll need to compile it. Long story-short:
> 
> 1. [Install rust](https://www.rust-lang.org/tools/install)(and git if you haven't).
> 2. `pip install git+https://github.com/axegon/spectrust.git`

I have already installed both the things. But exe file of rust is not being executed properly.

**Comment by axegon** [comment]
Windows? I seriously doubt the project would run under windows...

**Comment by KarnikaSaxena011** [comment]
> Windows? I seriously doubt the project would run under windows...

Yes sir i am running on windows...but when i use spectrust on google colab it works perfectly fine....it does not work on vs code.

**Comment by axegon** [comment]
Google colab is running within the context of some type of container(docker, rkt or something else) under linux, hence the reason it's working fine. That in addition to the fact that you're likely installing it directly from pypi over there and the package I've built and published there has been built for linux. Vscode or whatever you are using under windows is, as implied, windows. Even if you fix your rust installation and manage to compile it, it still likely won't work for multiple reasons. Feel free to try but I'd say the chances of it working are slim to none. The way I see it, you have 3 options:

1. Fork the project and figure out a way to make it work under windows.
2. Build a custom docker container and run it inside it.
3. Install a unix-based vm and use it inside.
4. Use a unix-based os.

**Comment by KarnikaSaxena011** [comment]
Ok sir...thanks for your help.
